.github/workflows/release.yml

```
@@ -51,22 +58,41 @@ jobs:
         roleId: ${{ secrets.VAULT_ROLE_ID }}
         secretId: ${{ secrets.VAULT_SECRET_ID }}
 
-    - name: Build Profiler Docker Image
-      continue-on-error: true #continue for now until we see it working in action
-      run: |
-        docker build . -t docker.elastic.co/observability/apm-agent-dotnet:${{ steps.bootstrap.outputs.agent-version }} \
-            --build-arg AGENT_ZIP_FILE=build/output/elastic_apm_profiler_${{ steps.bootstrap.outputs.agent-version }}-linux-x64.zip
-    - name: Push Profiler Docker Image
+    - name: Build and Push Profiler Docker Image
+      id: push
       continue-on-error: true #continue for now until we see it working in action
-      run: |
-        for i in $(seq 1 3); do [ $i -gt 1 ] && sleep 15; docker push docker.elastic.co/observability/apm-agent-dotnet:${{ steps.bootstrap.outputs.agent-version }} && s=0 && break || s=$?; done; (exit $s)
-        
+      uses: docker/build-push-action@v5.3.0
+      with:
```

```suggestion
      with:
        cache-from: type=gha
        cache-to: type=gha,mode=max
```

Unrelated optimization I found recently

---

Source: https://docs.docker.com/build/cache/backends/gha/#using-dockerbuild-push-action

---

ah!! I somehow it was some kind of GitHub specific registry but it's not it's about caching in GitHub actions. Thanks :)